src/main/java/org/eclipse/microprofile/starter/rest/APIService.java

```diff
@@ -0,0 +1,248 @@
+import org.eclipse.microprofile.starter.core.model.JessieSpecification;
+import org.eclipse.microprofile.starter.core.model.MicroProfileVersion;
+import org.eclipse.microprofile.starter.core.model.ModelManager;
+import org.eclipse.microprofile.starter.core.model.OptionValue;
+import org.eclipse.microprofile.starter.log.LoggingTask;
+import org.eclipse.microprofile.starter.rest.model.MPOptionsAvailable;
+import org.eclipse.microprofile.starter.rest.model.Project;
+import org.eclipse.microprofile.starter.view.EngineData;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.core.EntityTag;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+@RequestScoped
+public class APIService {
+
+    private static final Logger LOG = Logger.getLogger(APIService.class.getName());
+
+    private Map<MicroProfileVersion, MPOptionsAvailable> mpvToOptions;
+
+    private String readme;
+    private EntityTag readmeEtag;
+
+    @PostConstruct
+    public void init() {
+        mpvToOptions =
+                new HashMap<MicroProfileVersion, MPOptionsAvailable>(MicroProfileVersion.values().length) {
+                    {
+                        Stream.of(MicroProfileVersion.values()).forEach(mpv -> put(mpv, new MPOptionsAvailable(
+                                Stream.of(SupportedServer.values())
+                                        .filter(v -> v.getMpVersions().contains(mpv)).collect(Collectors.toList()),
+                                Stream.of(MicroprofileSpec.values())
+                                        .filter(v -> v.getMpVersions().contains(mpv)).collect(Collectors.toList()))));
+
+                    }
+
+                };
+        try {
+            Scanner s = new Scanner(FilesLocator.class.getClassLoader().getResourceAsStream("/REST-README.md")).useDelimiter("\\A");
+            readme = s.hasNext() ? s.next() : "";
+            readmeEtag = new EntityTag(Integer.toHexString(readme.hashCode()));
+        } catch (Exception e) {
+            e.printStackTrace();
+            LOG.log(Level.SEVERE, e.getMessage());
+        }
+    }
+
+    public static final String ERROR001 = "{\"error\":\"supportedServer query parameter is mandatory\"}";
+    public static final String ERROR002 = "{\"error\":\"Selected supportedServer is not available for given mpVersion\"}";
+    public static final String ERROR003 = "{\"error\":\"One or more selectedSpecs is not available for given mpVersion\"}";
```

Comment:
Returning the error code too will be useful for programmatic access.

Reply:
Fixed.